### PR TITLE
Added contribution data on search view and few test case using hypothesis

### PR DESCRIPTION
**Pull request by dhruvilpatel**

### CODE REVIEW

File: catalog/core/tests/test_views.py [hunk 1]
```diff
@@ -132,9 +155,11 @@ def test_canonical_publication_detail_view(self):
...
     # Test that publication detail is saved successfully or not
     def test_publication_detail_save_with_all_valid_fields(self):
-        management.call_command('zotero_import', test=True)
-        p = Publication.objects.first()
-        url = self.reverse('citation:publication_detail', query_parameters={'format': 'json'},
+        p = models(Publication, container=models(Container),
+                   added_by=models(User),
+                   code_archive_url=st.text(),
+                   url=st.text(), zotero_key=st.just(None)).example()
+        url = self.reverse(PUBLICATION_DETAIL_URL, query_parameters={'format': 'json'},
```
Comment (alee) [inline catalog/core/tests/test_views.py]: the slug is a good candidate for st.text()

File: catalog/core/tests/test_views.py [hunk 2]
```diff
@@ -207,86 +266,75 @@ def test_contact_view(self):
...
         json_data = json.loads(response.data['json'])
-        security_hash = json_data['security_hash']
-        timestamp = json_data['timestamp']
-
-        time.sleep(5)
-        response = self.post(url, {'name': 'John Watson', 'email': 'john@watson.com',
-                                   'message': 'Sherlock want to use this application.',
+        security_hash = json_data['security_hash'] + 'Fake'
+        timestamp = float(json_data['timestamp']) + 1
+        user = models(User).example()
```
Comment (cpritcha) [inline catalog/core/tests/test_views.py]: `models(User).example()` should be in the given decorator as `GENERATE_USER`. `GENERATE_USER` should look something like

```
DEFAULT_ALPHABET = st.characters(whitelist_categories={'Lu', 'Ll', 'Lt', 'Lm', 'Lo'}, blacklist_characters='\x00')

def text(min_size=1, max_size=20):
                  return st.text(alphabet=DEFAULT_ALPHABET, min_size=min_size, max_size=max_size)

GENERATE_USER = models(User,
                  username=st.just(str(username)),
                  email=st.just(str(username) + "@comses.net"),
                  first_name=text(),
                  last_name=text(),
                  password=st.just('')) 
```

(Adapted from `core.comses.net`)

File: catalog/core/tests/test_views.py [hunk 3]
```diff
@@ -55,28 +67,48 @@ def test_logout(self):
...
+        self.assertEqual(200, response.status_code)
+
+    # Test for invalid update of username
+    @settings(max_examples=30, perform_health_check=False)
+    @given(st.text())
+    def test_profile_invalid_update(self,username):
         url = self.reverse(USER_PROFILE_URL, query_parameters={'format': 'json'})
         self.login()
-        response = self.post(url, {'first_name': 'Test', 'last_name': 'Test'})
+        user_info = models(User).example()
```
Comment (cpritcha) [inline catalog/core/tests/test_views.py]: This should be in the given decorator as GENERATE_USER. See the comment below for detail.

File: catalog/core/tests/test_views.py [hunk 4]
```diff
@@ -154,41 +177,77 @@ def test_canonical_publication_detail_view(self):
...
-        p = models(Publication, container=models(Container),
-                   added_by=models(User),
-                   code_archive_url=st.text(),
-                   url=st.text(), zotero_key=st.just(None)).example()
+    @settings(max_examples=15, perform_health_check=False)
+    @given(GENRATE_PUBLICATION, st.text(min_size=10), st.sampled_from(GET_PLATFORM), st.sampled_from(GET_SPONSORS),
+           st.sampled_from(GET_MODEL_DOCUMENTATION), st.sampled_from(GET_STATUS), GENERATE_NOTES, st.booleans())
+    def test_publication_detail_save_with_all_valid_fields(self, p, slug, platforms, sponsors, model_documentations, status, notes,
+                                                           flagged):
+        user = models(User).example()
```
Comment (cpritcha) [inline catalog/core/tests/test_views.py]: This should be in the given decorator as GENERATE_USER. See the comment below for detail.

File: catalog/core/tests/test_views.py [hunk 5]
```diff
@@ -207,86 +266,75 @@ def test_contact_view(self):
...
-        time.sleep(5)
-
-        response = self.post(url, {'name': 'John Watson', 'email': 'john@watson.com',
-                                   'message': 'Sherlock want to use this application.',
-                                   'timestamp': timestamp,
-                                   'security_hash': security_hash,
-                                   'contact_number': ''})
-        self.assertTrue(400, response.status_code)
+        timestamp = float(json_data['timestamp'])
+        user = models(User).example()
```
Comment (cpritcha) [inline catalog/core/tests/test_views.py]: This should be in the given decorator as GENERATE_USER. See the comment below for detail.

File: catalog/core/tests/test_views.py [hunk 6]
```diff
@@ -55,28 +67,48 @@ def test_logout(self):
...
 class ProfileViewTest(BaseTest):
     def test_profile_view(self):
-        self.without_login_and_with_login_test('core:user_profile')
+        self.without_login_and_with_login_test(USER_PROFILE_URL)
 
+    @settings(max_examples=30, perform_health_check=False)
     def test_profile_update(self):
         url = self.reverse(USER_PROFILE_URL, query_parameters={'format': 'json'})
         self.login()
+        user_info = models(User).example()
```
Comment (cpritcha) [inline catalog/core/tests/test_views.py]: This should be in the given decorator as GENERATE_USER. See the comment below for detail.

File: catalog/core/tests/test_views.py [hunk 7]
```diff
@@ -17,6 +17,18 @@
 WORKFLOW_URL = 'core:curator_workflow'
 SENT_INVITES_URL = 'core:send_invites'
 
+GET_TAGS = list(Tag.objects.all().values_list('name'))
+GET_PLATFORM = list(Platform.objects.all().values_list('name'))
+GET_SPONSORS = list(Sponsor.objects.all().values_list('name'))
+GET_MODEL_DOCUMENTATION = list(ModelDocumentation.objects.all().values_list('name'))
+GET_STATUS = [s[0] for s in Publication.Status]
+GENRATE_PUBLICATION = models(Publication, container=models(Container),
```
Comment (cpritcha) [inline catalog/core/tests/test_views.py]: GENRATE_PUBLICATION -> GENERATE_PUBLICATION

**Pull request merged**